### ISSUE TITLE
Exercise 7 of 13: Promise after promise

### ISSUE THREAD

**Issue by probil** [stevekane/promise-it-wont-hurt#88]
Issues I have:
1) Task description is abstract and it's hard to realize what you should do here.
2) Solution should be executed in 0.4s-0.6s. But my solution executed in ~0.1s and I can't pass it.
```
✗ Expected execution time: 546±100ms. Submission execution time: 129ms.
```
O_o
There is no mention about this restriction in description. 
3) Where can I get 'secret value'?
```
                 ACTUAL                                 EXPECTED                
────────────────────────────────────────────────────────────────────────────────

   "[Function]"                        !=    "ef7062436a06faf36d1280b005347dd50aad52c3"
```
It's a magic, nothing more... 

This task rather for improving skills in reading developers thoughts rather than learning Promises.

**Comment by heymark** [comment]
Agreed, this is obtuse, and inconsistent with the style of the exercises up to this point. No mention is made of the fact that this exercise relies on interacting with the test suite to obtain the "secret value" – the "secret value" in a previous exercise was a string literal, and previous exercises have stood on their own (i.e. been testable outside of the workshopper environment).

Could `first` and `second` be included in a skeleton (as has been done for previous exercises) to make it clear that writing them isn't part of the assignment, and to make the whole thing testable outside of the workshopper?

Alternatively, perhaps make it explicit that `first` and `second` exist elsewhere, similar to Exercise 9's `Some invalid JSON will be available on process.argv[2]`.

**Comment by TimothyGu** [comment]
I modeled this specific exercise after [learnyounode](https://github.com/workshopper/learnyounode/blob/master/exercises/http_uppercaserer/exercise.js#L8-L10)'s use random word generator, with the obvious intention to make it harder to cheat the system. Now, I guess the instructions can be improved, and thanks go to @heymark for pointing out some obvious places.

> Could `first` and `second` be included in a skeleton (as has been done for previous exercises) to make it clear that writing them isn't part of the assignment, and to make the whole thing testable outside of the workshopper?

Many learnyounode exercises aren't really testable outside the workshopper, but I can definitely provide a skeleton.

> Alternatively, perhaps make it explicit that `first` and `second` exist elsewhere, similar to Exercise 9's `Some invalid JSON will be available on process.argv[2]`.

I'd probably go against this, since `first` and `second` are words, and won't work as well as `process.argv[2]`.

**Comment by JoeDoyle23** [comment]
We just ran into this issue as well at the Node School held during Node Summit. I think the approach is ok, but the instructions are very unclear that the `first` and `second` functions will be injected. Would you be open to a PR for an update to the instructions for this one?

**Comment by TimothyGu** [comment]
@JoeDoyle23, yes, of course.

**Comment by probil** [comment]
@JoeDoyle23 You are right. Approach is good but current exercise is unclear. I have finished first 6 exercises without any issues

**Comment by donhenton** [comment]
not even getting a value for process.argv[2], either in run or verify mode.  Insert sad-face here.

**Comment by adekbadek** [comment]
@TimothyGu #93 is resolving this issue, could you please merge it?

**Comment by TimothyGu** [comment]
@adekbadek, comments for #93 added.

**Comment by probil** [comment]
It seems like #93 resolves my issue. The exercise description is not so "magic" as before.

**Comment by mn113** [comment]
PR #93 is a great improvement to the wording! Now if someone would kindly merge it in, the rest of the world can stop banging their heads against this frustrating exercise...

**Comment by sdlambert** [comment]
I can second all of the previous users here, the wording is not clear and this PR would help tremendously. Thanks!

**Comment by Chrinkus** [comment]
THIS. Very fun and well designed workshop up until this point.

**Comment by jfisher446** [comment]
@TimothyGu Could we have another look at merging #93 to solve this issue? Thank you!
I'll echo the others above that it was all smooth sailing until this hard blocker due to confusion over the task instructions.

**Comment by jjhampton** [comment]
I agree with the comments above - I was only able to comprehend the instructions enough to complete this exercise after #93 was added.

**Comment by hotdang-ca** [comment]
Oh good, i'm not the only one. I rage-quit during this one...
